packages/chat/components/chat/src/chat.ts

```diff
@@ -0,0 +1,355 @@
+    this._messageText = event.target.value;
+  }
+
+  /** get time of message formatted as 1:23pm or 4:56am
+   **/
+  _getCurrentTime(): string {
+    const now = new Date();
+    let hours: number = now.getHours();
+    let minutes: number = now.getMinutes();
+    let amOrpm: string = hours >= 12 ? 'pm' : 'am';
+    
+    hours = hours % 12 || 12;
+    const formattedMinutes: string = minutes < 10 ? '0' + minutes : minutes.toString();
+
+    const currentTime: string = hours + ':' + formattedMinutes + amOrpm;
+
+    return currentTime;
+  }
+
+  /** handle hover event on any message, open up feedback buttons
+   * @param{Number} index of message within the messages array
+   * @param{event} lit hover event
+   * @param{type} type of message being hovered upon
+   **/
+  _handleHoverIn(index, event, type){
+    console.log(index);
+    let object = event.target;
+    if(object.className == "message-container"){
+      let dropdown = object.parentElement.querySelector("."+type+"-dropdown");
+      dropdown.style.height = "38px";
+      //this.requestUpdate();
+    }
+  }
+
+  /** handle hover event on any message, open up feedback buttons
+   * @param{Number} index of message within the messages array
+   * @param{event} lit hover event
+   * @param{type} type of message being hovered upon
+   **/
+  _handleHoverOut(index, event, type){
+    let object = event.target;
+    //let target = event.target.firstElementChild;
+    let relatedTarget = event.relatedTarget;
+    console.log(index);
+
+    if(relatedTarget && relatedTarget.className !== "user-dropdown"
+      && relatedTarget.className !== "bot-dropdown"
+      && relatedTarget.className !== "message-container"
+      && relatedTarget.className !== "message-icon"
+      && relatedTarget.className !== "message-text"
+      && relatedTarget.className !== "small-button"){
+      let dropdown = object.parentElement.querySelector("."+type+"-dropdown");
+      dropdown.style.height = "0px";
+      //this.requestUpdate();
+    }
+  }
+
+  /** auto-scroll chat-messages div when a new message has appeared
+   **/
+  _updateScroll(){
+    let scrollDiv = this.shadowRoot?.querySelector(".chat-messages");
+    setTimeout(() =>{
+      scrollDiv?.scrollTo({
+        top: scrollDiv?.scrollHeight,
+        behavior: 'smooth'
+      });
+    }, 200);
+  }
+
+  /** editing function when a user click the edit button
+   * @param{event} lit click event
+   * @param{index} selected message index within the messages array
+   **/
+  _handleEdit(event,index){
+    let parent = event.target.parentElement;
+    let textDiv = parent.querySelector(".message-text");
+    console.log(parent);
+    console.log(textDiv);
+    console.log(index);
+
+    /*
+    let text = parent.querySelector(".message-text").innerHTML;
+    console.log(text)
+
+    const textArea = document.createElement('textarea');
+    textArea.className = 'editableTextArea';
+    textArea.value = text;
+
+    textDiv.replaceWith(textArea)*/
```

Comment:
Do we need this commented out section?

Reply:
it was an early version of text editing but unneeded since Priyanshu put in the text feedback component, removed it